recipe_batches/recipe_batches.py

```diff
@@ -1,31 +1,41 @@
 #!/usr/bin/python
 
 import math
+import operator
+
 
 def recipe_batches(recipe, ingredients):
-  whole_batches = {}
-  for rkey in recipe.items():
-  
-    for ikey in ingredients.items():
+    # get the maximum value from ingredients.items, so we have a starting point
+    batches = max(ingredients.items(), key=operator.itemgetter(1))[1]
+    # loop through the recipes dict
+    for rkey in recipe.items():
+        # if the key from recipes does NOT appear in ingredients, batches = 0 because
+        # we cannot make the recipe
+        if rkey[0] not in ingredients.keys():
+            batches = 0
+        # loop through the ingredients dict
+        for ikey in ingredients.items():
```

Comment:
i ran print statements on this, and this runs 9 times! O(n**2). 
remember this are dictionaries, so if you loop through `for rkey in recipe.items()`, you can use `rkey` as a key to the dictionary (javascript object)